build/build.go

```diff
@@ -590,6 +592,15 @@ func toSolveOpt(ctx context.Context, di DriverInfo, multiDriver bool, opt Option
 		}
 	}
 
+	if len(opt.Attests) > 0 {
+		if !bopts.LLBCaps.Contains(apicaps.CapID("exporter.image.attestations")) {
+			return nil, nil, errors.Errorf("attestations are not supported by the current buildkitd")
+		}
+	}
```

Comment:
Don't think it's necessary to check caps.

Reply:
IMO we should - since they're added to the frontend opts, old backends will silently ignore them.

We need to make sure that if the user asks for attestations, they get them - I think it's close to a security issue if the user can get a silent failure if they don't get attestations when they've been requested.